python/pyarrow/tests/test_schema.py

```diff
@@ -564,7 +564,8 @@ def test_schema_from_pandas():
             '2007-07-13T01:23:34.123456789',
             '2006-01-13T12:34:56.432539784',
             '2010-08-13T05:46:57.437699912'
-        ], dtype='datetime64[ns]')
+        ], dtype='datetime64[ns]'),
+        pd.array([1, 2, None], dtype=pd.Int32Dtype()),
```

Comment:
This probably a conditional version check (pd.array only exists in recent pandas versions, so only add it to `inputs` if pandas >= 0.24